12-watchtower-API.md

```diff
@@ -44,41 +46,37 @@ Connections between the client and the server can be long-lived or restarted for
 
 ### The `wt_init` message
 
-This messsage contains the information about a node and the type of appointments he is willing to create / accept.
+This message contains the information about a node and the type of appointments he is willing to create / accept.
 
-1. type: 128 (`wt_init`)
+1. type: ? (`wt_init`)
 2. data:
-   * [`u16`:`client_server`]
    * [`u16`:`aclen`]
    * [`aclen*bytes`:`accepted_ciphers`]
    * [`u16`:`modlen`]
    * [`modlen*bytes`:`modes`]
    * [`u16`:`qoslen`]
 	* [`qoslen*bytes`:`qos`]	
 
-`client_server` defines whether the sender is a client or a server.
-
-`accepted_ciphers` define the ciphers that the sender accepts and that he can use to encrypt / decrypt data. Accepted cyphers include `chacha20` and `aes-gcm-256`.
+`accepted_ciphers` define the ciphers that the sender implements and that he can use to encrypt / decrypt data. Accepted cyphers include `chacha20` and `aes-gcm-256`.
 
-`modes` define the operation mode requested / accepted. Modes include `altruistic` and `non-altuistic`.
+`modes` define the operation mode requested / accepted. Modes include `altruistic` and `non-altruistic`.
 
 `qos` defines whether the sender is requesting / accepting Quality of Service for his appointments. The only QoS offered at the moment is `accountability`.
 
 #### Requirements
 The sending node: 
 
 * MUST send `wt_init` as the first message.
-* MUST set `client_server` to the corresponding value, depending on whether it is a client or a server.
 * MUST set `accepted_ciphers` to the list of ciphers he implements.
 * MUST set `modes` to the list of modes that he is willing to accept.
 * SHOULD set `qos` to the the quality of service he is requesting / offering. 
+* upon receiving `wt_init` from a node he has not started a handshake with:
+	* MUST fail the connection
 
 The requesting node: 
 
 * MUST receive `wt_init` before sending any other message.
-* MUST respond with its own `wt_init` message
-* upon receiving a `client_server` field that matches its own (a server receives a `wt_init` from a server or a client receives one from a client):
-	* MUST fail the connection
+* MUST respond with its own `wt_init` message.
 * upon receiving an `accepted_ciphers` that does not contain any of its own accepted ciphers:
 	* MUST fail the connection
 * upon receiving a `modes` that does not contain any of its own accepted modes:
```

Comment:
It shouldn't matter. Also, as specified in the fixme, it'll be better to have a node discovery instead of a wt_init. Leaving this like this for the draft since I'd like to see what the community think about the two approaches.

Reply:
i'm actually thinking. 

the init message should really just be the client asking "what do you offer" and the server replying with a bunch of options.

why does the client need to indicate what type of service they care about? really they just do this discovery / init message to work out what is on offer.

Reply:
I sort of agree. The user may just ping the tower for info about the services and the tower may respond with what it is offering. I'll change it to fit that. It also resembles more a peer discovery.

Reply:
kk